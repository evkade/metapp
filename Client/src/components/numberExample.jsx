import React from "react";

export const numberExample = (renderNumber, myNumber) => {
  return renderNumber ? <div> {myNumber}</div> : <div> Random Text </div>;
};
