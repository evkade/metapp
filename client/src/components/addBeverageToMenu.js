import React from "react";

export const AddBeverageToMenu = ({searchBeverage, cocktailPromise, searchResult, searchError}) => {
  const [query, setQuery] = React.useState("");
  console.log(searchResult);

  return (
    <div>
      <input
        onChange={() => setQuery(event.target.value)}
      ></input>
      <button type="submit" onClick={() => searchBeverage(query)}>
        {" "}
        Search{" "}
      </button>
      <div>{searchResult ? searchResult.drinks.map(cocktail => <div key={cocktail.idDrink}>{cocktail.strDrink}</div>) : null}</div>
    </div>
  );
};
