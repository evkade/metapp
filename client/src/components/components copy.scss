@import "variables.scss";
@import "../../node_modules/bootstrap/scss/bootstrap.scss";
@import "mixins.scss";

// customizeMenu => admin-menu-container (customizeMenuPresenter)
//AdminViewDrinkOrder
//BeverageCard
//creatrbeverageformenumodal
// customizeMenu

@include keyframe__neon--flicker(
  flicker-cerise,
  white,
  white,
  $neon-color-pink
);
@include keyframe__neon--flicker(
  flicker-green,
  black,
  $neon-color-green,
  $neon-color-green
);

@include keyframe__neon--flicker(
  flicker-yellow,
  black,
  $neon-color-yellow,
  $neon-color-yellow
);

body {
  margin: 0px !important;
  font-family: $general-font;
}

#app {
  height: 100vh;
  overflow: hidden;
  display: flex;
  flex-direction: column;
}

.user-form {
  height: 100vh !important;
  @include alignedGeneralFont();
  @at-root #{&}__card {
    background: rgb(240, 240, 240);
    background-image: linear-gradient(to left, rgb(240, 240, 240), #ffff);
    border-radius: 6px;
    flex-direction: column;
    padding: 10px;
    box-shadow: 5px 5px 8px;
    width: 90%;
    max-width: 600px;
    aspect-ratio: 5/3;
    @include centerVerticalHorizontal();
    align-items: unset;
  }
  @at-root #{&}__flex-row {
    display: flex;
    height: 30%;
  }
  @at-root #{&}__logo {
    flex: 0.5;
    @include centerVerticalHorizontal();
  }
  @at-root #{&}__logo img {
    height: 100%;
  }
  @at-root #{&}__title {
    flex: 0.5;
    font-size: 40px;
    @include centerVerticalHorizontal();
  }
  @at-root #{&}__errElement {
    height: 50px;
    padding: 20px;
    color: red;
  }
}

.container--centered {
  @include centerVerticalHorizontal();
}

.input-container {
  flex: 1;
  display: grid;
  font-family: $general-font;
  @at-root #{&}__element {
    border: 1px solid black;
    border-radius: 6px;
    margin: 4px;
    padding: 10px;
  }
}

.container--general {
  @include brickBackground();
}

.entry-view {
  height: 100vh !important;
  @at-root #{&}__neon-text {
    @include pageTitleFont();
    font-size: 20vw;
    font-family: $barcode-font;
    margin-bottom: 10vh;
    max-height: 300px;
  }
  @at-root #{&}__logos img {
    height: 8vh;
    margin-bottom: 2vh;
    margin-left: 10px;
    margin-right: 10px;
  }
}

.neon-button {
  background: rgba(0, 0, 0, 0.6);
  font-family: $button-entryView-font;
  font-size: 5vw;
  margin: 20px;
  width: 5em;
  padding: 2vh;
  border-radius: 25px;
  @at-root {
    .neon-button--green {
      color: $neon-color-green;
      border: 3px solid $neon-color-green;
      animation: flicker-green 8s infinite alternate;
    }
    .neon-button--yellow {
      color: $neon-color-yellow;
      border: 3px solid $neon-color-yellow;
      animation: flicker-yellow 10s infinite alternate;
    }
  }
}

@media screen and (prefers-reduced-motion) {
  .entry-view__neon-text,
  .neon-button--green,
  .neon-button--yellow {
    animation: none !important;
  }
}

.card-drink {
  display: inline-block;
  border: 1px solid black;
  border-radius: 10%;
  margin: 10px;
  padding: 10px;
  background-color: $light-neon-color-pink;
  .card-drink--made {
    background-color: $light-neon-color-yellow;
  }
  .card-drink--finished {
    background-color: $light-neon-color-green;
  }
}

.drink-list {
  width: 100%;
  flex: 1;
  @at-root #{&}__container {
    height: 80%;
    width: 90%;
    max-width: 800px;
    overflow-y: scroll;
    background-color: black;
    border-radius: 6px;
    flex: 1;
    margin-bottom: 20px;
  }
  @at-root #{&}__button {
    z-index: 1;
    position: absolute;
    bottom: 30px;
    border-radius: 6px;
    font-family: "Pacifico", cursive;
    font-size: 3vmax;
    width: 70%;
    border: 1px solid black;
    background: rgb(240, 200, 48);
  }
  @at-root #{&}__row {
    height: 15%;
    display: flex;
    align-items: center;
    margin: 10px;
    @include alignedGeneralFont();
  }
  @at-root #{&}__image {
    height: 100%;
    border-radius: 6px;
    border: 1px solid white;
  }
  @at-root #{&}__column--flexed {
    flex: 1;
    color: white;
  }
  @at-root #{&}__buttons {
    flex-direction: row;
    flex: 1;
    @include centerVerticalHorizontal();
    font-family: $general-font;
    color: white;
  }
  @at-root #{&}__star {
    color: lightgray;
    margin-right: -20px;
    margin-top: 5%;
    z-index: 1;
  }
  @at-root #{&}__star--active {
    color: rgb(240, 200, 48);
  }
}

.title-neon--big {
  font-size: 10vh;
  animation: flicker-cerise 10s infinite alternate;
  @include pageTitleFont();
}

.title-neon--small {
  font-size: 7vh;
  animation: flicker-cerise 10s infinite alternate;
  @include pageTitleFont();
}

.modal {
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  @include centerVerticalHorizontal();
  @at-root #{&}__content {
    width: 500px;
    background-color: #fff;
  }
  @at-root #{&}__headfoot {
    padding: 10px;
  }
  @at-root #{&}__title {
    margin: 0;
  }
  @at-root #{&}__body {
    padding: 10px;
    border-top: 1px solid #eee;
    border-bottom: 1px solid #eee;
  }
}

.general-button--bw {
  background: none;
  margin: 5px;
  color: white;
  border-radius: 6px;
  border: 1px solid white;
}

.general-button--bw:disabled {
  color: grey;
  border: 1px solid grey;
}

.profile-view {
  height: 100vh !important;
}

.info-card-drink {
  background: black;
  color: white;
  width: 80%;
  height: 20%;
  border-radius: 6px;
  padding: 6px;
  margin-bottom: 10px;
  @at-root #{&}__row {
    flex-direction: row;
    @include centerVerticalHorizontal();
    border-bottom: 1px solid white;
    padding: 10px 0px;
  }
  @at-root #{&}__column {
    text-align: left;
    flex: 1;
  }
  @at-root #{&}__column--flex {
    flex: 1;
    @include centerVerticalHorizontal();
  }
  @at-root #{&}__container--scroll {
    height: 70%;
    overflow-y: scroll;
  }
  @at-root #{&}__title {
    font-size: 20px;
  }
}

.admin-menu-container {
  background-color: lightgrey;
  height: 100%;
  input {
    border: 1px solid black;
    border-radius: 5px;
    padding-top: 15px;
    width: 60%;
    padding: 2px 2px 2px 2px;
    outline: 0;
    background-color: #f5f5f5;
  }
  input:hover,
  input:focus {
    border: 1.5px solid grey;
    background-color: grey;
  }
  @at-root #{&}__row {
    display: grid;
    grid-column-gap: 5px;
  }
  @at-root #{&}__column {
    background-color: black;
    border-radius: 6px;
    grid-row-start: 1;
    grid-row-end: 2;
    grid-column-start: 2;
    grid-column-end: 2;
  }
  @at-root #{&}__column--flex {
    grid-column-start: 1;
    grid-column-end: 1;
  }
}
