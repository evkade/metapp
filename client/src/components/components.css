:root {
  --neon-color-pink: #ee2a7b;
  --light-neon-color-pink: #fbd0e2;
  --neon-color-green: #2aee4b;
  --light-neon-color-green: #d0fbd7;
  --neon-color-yellow: #deee2a;
  --light-neon-color-yellow: #f8fbd0;
  --dark-background: rgb(26, 26, 26);
  --barcode-font: "Libre Barcode 39 Extended Text", cursive;
  --button-neon-font: "Pacifico", cursive;
  --general-font: "Montserrat", sans-serif;
}

body {
  margin: 0px !important;
}

.signInForm {
  text-align: center;
  font-family: var(--general-font);
}

.signInForm__form {
  background: rgb(240, 240, 240);
  background-image: linear-gradient(to left, rgb(240, 240, 240), #ffff);
  border-radius: 6px;
  display: flex;
  flex-direction: column;
  padding: 10px;
  justify-content: center;
  box-shadow: 5px 5px 8px;
  width: 90%;
  max-width: 600px;
  aspect-ratio: 5/3;
}

.signInForm__form--flexRow {
  display: flex;
  height: 30%;
  margin-bottom: 20px;
}

.signInForm__logo {
  flex: 0.5;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.signInForm__title {
  display: flex;
  flex: 0.5;
  align-items: center;
  font-size: 40px;
}

.signInForm__logo img {
  height: 100%;
}

.signInForm__inputContainer {
  flex: 1;
  display: grid;
}

div.signInForm__inputContainer input,
div.signInForm__inputContainer button {
  font-size: 20px;
  padding: 5px;
  font-family: var(--general-font);
}

.signInForm__inputContainer--input {
}

.signInForm__errElement {
  padding: 20px;
  color: red;
}

.signInForm__successElement {
  display: block;
  color: white;
}

.red-brick {
  color: red;
  width: 50px;
  height: 50px;
}

.entryView {
}

.brickBackground {
  background: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.8)),
    url("https://i.imgur.com/mnVAkgM.jpg");
  display: flex;
  background-repeat: no-repeat;
  background-size: cover;
  justify-content: center;
  align-items: center;
  height: 100vh;
  flex-direction: column;
}

.entryView__neonText {
  color: white;
  font-size: 20vw;
  font-family: var(--barcode-font);
  --neon-color: var(--neon-color-pink);
  --neon-color-close1: white;
  --neon-color-close2: white;
  animation: flicker 1.5s infinite alternate;
  box-shadow: none !important;
  margin-bottom: 10vh;
}

@keyframes flicker {
  0%,
  18%,
  22%,
  25%,
  53%,
  57%,
  100% {
    text-shadow: 0 0 3px var(--neon-color-close1),
      0 0 10px var(--neon-color-close2), 0 0 21px var(--neon-color),
      0 0 42px var(--neon-color), 0 0 82px var(--neon-color),
      0 0 92px var(--neon-color), 0 0 102px var(--neon-color),
      0 0 111px var(--neon-color);
    box-shadow: 0px 0px 10px 1px var(--neon-color);
  }
  20%,
  20.1% {
    text-shadow: none;
    box-shadow: none;
  }
}

.entryView__buttons {
  background: rgba(0, 0, 0, 0.6);
  font-family: var(--button-neon-font);
  font-size: 5vw;
  margin: 20px;
  width: 4em;
  padding-bottom: 2vh;
  border-radius: 25px;
}

.entryView__buttons--green {
  color: var(--neon-color-green);
  border: 3px solid var(--neon-color-green);
  --neon-color: var(--neon-color-green);
  --neon-color-close1: black;
  --neon-color-close2: var(--neon-color-green);
  animation: flicker 8s infinite alternate;
}

.entryView__buttons--yellow {
  color: var(--neon-color-yellow);
  border: 3px solid var(--neon-color-yellow);
  --neon-color: var(--neon-color-yellow);
  --neon-color-close1: black;
  --neon-color-close2: var(--neon-color-yellow);
  animation: flicker 10s infinite alternate;
}

@media screen and (prefers-reduced-motion) {
  .entryView__neonText,
  .entryView__buttons--green,
  .entryView__buttons--yellow {
    animation: none !important;
  }
}

.entryView__logos img {
  height: 8vh;
  margin-bottom: 10vh;
  margin-left: 10px;
  margin-right: 10px;
}

.drinkCard {
  display: inline-block;
  border: 1px solid black;
  border-radius: 10%;
  margin: 10px;
  padding: 10px;
  background-color: var(--light-neon-color-pink);
}

.drinkCard--made {
  background-color: var(--light-neon-color-yellow);
}

.drinkCard--finished {
  background-color: var(--light-neon-color-green);
}
