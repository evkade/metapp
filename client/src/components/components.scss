@import "variables.scss";
@import "../../node_modules/bootstrap/scss/bootstrap.scss";
@import "mixins.scss";

@include keyframe__neon--flicker(
  flicker-cerise,
  white,
  white,
  $neon-color-pink
);

@include keyframe__neon--flicker(
  flicker-green,
  black,
  $neon-color-green,
  $neon-color-green
);

@include keyframe__neon--flicker(
  flicker-yellow,
  black,
  $neon-color-yellow,
  $neon-color-yellow
);

body {
  margin: 0px !important;
  font-family: $general-font;
}

#app {
  height: 100vh;
  overflow: hidden;
  display: flex;
  flex-direction: column;
}

.user-form {
  height: 100vh !important;
  @include alignedGeneralFont();
  @at-root #{&}__card {
    background: rgb(240, 240, 240);
    background-image: linear-gradient(to left, rgb(240, 240, 240), #ffff);
    border-radius: 6px;
    flex-direction: column;
    padding: 10px;
    box-shadow: 5px 5px 8px;
    width: 90%;
    max-width: 600px;
    aspect-ratio: 5/3;
    @include centerVerticalHorizontal();
    align-items: unset;
  }
  @at-root #{&}__flex-row {
    display: flex;
    height: 30%;
  }
  @at-root #{&}__logo {
    flex: 0.5;
    @include centerVerticalHorizontal();
  }
  @at-root #{&}__logo img {
    height: 100%;
  }
  @at-root #{&}__title {
    flex: 0.5;
    font-size: 40px;
    @include centerVerticalHorizontal();
  }
  @at-root #{&}__errElement {
    height: 50px;
    padding: 20px;
    color: red;
  }
}

.beverage-form {
  display: flex;
  flex-direction: column;
}

.container--centered {
  @include centerVerticalHorizontal();
  height: 100%;
  width: 100%;
}

.input-container {
  flex: 1;
  display: grid;
  font-family: $general-font;
  @at-root #{&}__element {
    border: 1px solid black;
    border-radius: 6px;
    margin: 4px;
    padding: 10px;
  }
}

.container--general {
  @include brickBackground();
}

.entry-view {
  height: 100vh !important;
  @at-root #{&}__neon-text {
    @include pageTitleFont();
    font-size: 20vw;
    font-family: $barcode-font;
    margin-bottom: 10vh;
    max-height: 300px;
  }
  @at-root #{&}__logos img {
    height: 8vh;
    margin-bottom: 2vh;
    margin-left: 10px;
    margin-right: 10px;
  }
}

.neon-button {
  background: rgba(0, 0, 0, 0.6);
  font-family: $button-entryView-font;
  font-size: 5vw;
  margin: 20px;
  width: 5em;
  padding: 2vh;
  border-radius: 25px;
  @at-root {
    .neon-button--green {
      color: $neon-color-green;
      border: 3px solid $neon-color-green;
      animation: flicker-green 8s infinite alternate;
    }
    .neon-button--yellow {
      color: $neon-color-yellow;
      border: 3px solid $neon-color-yellow;
      animation: flicker-yellow 10s infinite alternate;
    }
  }
}

@media screen and (prefers-reduced-motion) {
  .entry-view__neon-text,
  .neon-button--green,
  .neon-button--yellow {
    animation: none !important;
  }
}

.card-order {
  display: inline-block;
  border: 1px solid black;
  border-radius: 10%;
  margin: 10px;
  padding: 10px;
  background-color: $light-neon-color-pink;
  &--made {
    background-color: $light-neon-color-yellow !important;
    border: 1px solid $neon-color-yellow !important;
  }
  &--finished {
    background-color: $light-neon-color-green;
    border: 1px solid $neon-color-green;
  }
  &--cancelled {
    background-color: #cccccc;
  }
  &__text--title {
    cursor: pointer;
  }
  &__button {
    border-radius: 10px;
    padding: 5px;
    margin: 3px;
    background-color: white;
  }
}

.drink-list {
  width: 100%;
  flex: 1;
  @at-root #{&}__container {
    height: 80%;
    width: 90%;
    max-width: 800px;
    overflow-y: scroll;
    background-color: black;
    border-radius: 6px;
    flex: 1;
    margin-bottom: 20px;
  }
  @at-root #{&}__container--grey {
    background-color: lightgrey;
    height: 70%;
    max-height: 200px;
    overflow-y: scroll;
  }
  @at-root #{&}__container--grey-full {
    height: 100%;
    max-height: 100%;
    width: 100%;
  }
  @at-root #{&}__button {
    z-index: 1;
    position: absolute;
    bottom: 30px;
    border-radius: 6px;
    font-family: "Pacifico", cursive;
    font-size: 3vmax;
    width: 70%;
    border: 1px solid black;
    background: rgb(240, 200, 48);
  }
  @at-root #{&}__row {
    display: flex;
    align-items: center;
    margin: 10px;
    @include alignedGeneralFont();
  }
  @at-root #{&}__row--constrained {
    height: 15%;
  }
  @at-root #{&}__row--white {
    color: white;
    text-align: center;
  }
  @at-root #{&}__image {
    height: 100%;
    border-radius: 6px;
    border: 1px solid white;
  }
  @at-root #{&}__column {
    color: white;
    width: 30%;
  }
  @at-root #{&}__column--black {
    color: black;
  }
  @at-root #{&}__column--flexed {
    flex: 1;
  }
  @at-root #{&}__buttons {
    flex-direction: row;
    flex: 1;
    @include centerVerticalHorizontal();
    font-family: $general-font;
    color: white;
  }
  @at-root #{&}__star {
    color: lightgray;
    margin-right: -20px;
    margin-top: 5%;
    z-index: 1;
  }
  @at-root #{&}__star--active {
    color: rgb(240, 200, 48);
  }
}

.title-neon--big {
  font-size: 10vh;
  animation: flicker-cerise 10s infinite alternate;
  @include pageTitleFont();
}

.title-neon--small {
  font-size: 7vh;
  animation: flicker-cerise 10s infinite alternate;
  @include pageTitleFont();
}

.modal {
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  @include centerVerticalHorizontal();
  @at-root #{&}__content {
    width: 500px;
    background-color: #fff;
    label {
      margin: 5px;
    }
    input {
      height: 25px;
      margin: 5px;
      padding: 3px 5px;
      border-radius: 5px;
    }
    input:hover {
      background-color: lightgrey;
    }
    button {
      border-radius: 5px;
    }
  }
  @at-root #{&}__headfoot {
    padding: 10px;
  }
  @at-root #{&}__title {
    margin: 0;
  }
  @at-root #{&}__body {
    padding: 10px;
    border-top: 1px solid #eee;
    border-bottom: 1px solid #eee;
  }
}

.general-button--bw {
  background: none;
  margin: 5px;
  color: white;
  border-radius: 6px;
  border: 1px solid white;
}

.general-button--black {
  color: black;
  border: 1px solid black;
}

.general-button--bw:disabled {
  color: grey;
  border: 1px solid grey;
}

.profile-view {
  height: 100vh !important;
}

.info-card-drink {
  background: black;
  color: white;
  width: 80%;
  height: 20%;
  border-radius: 6px;
  padding: 6px;
  margin-bottom: 10px;
  @at-root #{&}__row {
    flex-direction: row;
    @include centerVerticalHorizontal();
    border-bottom: 1px solid white;
    padding: 10px 0px;
  }
  @at-root #{&}__column {
    text-align: left;
    flex: 1;
  }
  @at-root #{&}__column--flex {
    flex: 1;
    @include centerVerticalHorizontal();
  }
  @at-root #{&}__container--scroll {
    height: 70%;
    overflow-y: scroll;
  }
  @at-root #{&}__title {
    font-size: 20px;
  }
}

.admin-menu-container {
  background-color: black;
  height: 100%;
  padding: 10px;
  @at-root #{&}__button {
    background: none;
    margin: 5px;
    color: white;
    border-radius: 6px;
    border: 1px solid white;
  }
  @at-root #{&}__button:hover {
    background-color: grey;
  }
  @at-root #{&}__tabs {
    margin-left: 6px;
  }
  @at-root #{&}__tab {
    background: black;
    padding: 5px;
    color: white;
    display: inline-block;
    border: 1px solid white;
    margin-bottom: -1px;
    border-radius: 3px 3px 0px 0px;
  }
  @at-root #{&}__tab:disabled {
    background: lightgrey;
    color: black;
  }
  @at-root #{&}__row {
    display: flex;
    max-width: 100%;
    height: 70%;
  }
  @at-root #{&}__column {
    background-color: lightgray;
    flex: 1 0 45%;
    border-radius: 7px;
    margin-left: 3px;
    text-align: center;
    display: flex;
    justify-content: center;
    flex-direction: column;
    input {
      height: 30px;
      padding: 3px 5px;
      margin: 5px;
      border-radius: 5px;
      border: 1px solid black;
      background-color: #f5f5f5;
    }
    input:hover,
    select:hover,
    input:focus {
      border: 1.5px solid grey;
      background-color: grey;
    }
    select {
      height: 30px;
      padding: 3px 5px;
      margin: 5px;
      border-radius: 5px;
    }
  }
  @at-root #{&}__column--flex {
    border-radius: 0px 7px 7px 7px;
    margin-left: 6px;
    margin-right: 3px;
  }
  @at-root #{&}__subtitle {
    color: white;
  }
  @at-root #{&}__collapsible--collapsed {
    display: none !important;
  }
  @at-root #{&}__collapsible {
    height: 50%;
    display: block;
    overflow-y: scroll;
    border-radius: 6px;
    margin-bottom: 10px;
  }
}
.height100 {
  height: 100%;
}

.navbar {
  background-color: black;
}

.dropdown-toggle::after {
  color: white;
}

.dropdown-menu.show {
  background-color: black;
}

.dropdown-item:hover {
  background-color: black;
  opacity: 0.5;
}
