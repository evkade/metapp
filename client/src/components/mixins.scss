@mixin keyframe__neon--flicker(
  $name,
  $neon-color-close1,
  $neon-color-close2,
  $neon-color
) {
  @keyframes #{$name} {
    0%,
    18%,
    22%,
    25%,
    53%,
    57%,
    100% {
      text-shadow: 0 0 3px $neon-color-close1, 0 0 10px $neon-color-close2,
        0 0 21px $neon-color, 0 0 42px $neon-color, 0 0 82px $neon-color,
        0 0 92px $neon-color, 0 0 102px $neon-color, 0 0 111px $neon-color;
      box-shadow: 0px 0px 10px 1px $neon-color;
    }
    20%,
    20.1% {
      text-shadow: none;
      box-shadow: none;
    }
  }
}

@mixin centerVerticalHorizontal {
  display: flex;
  justify-content: center;
  align-items: center;
}

@mixin alignedGeneralFont {
  text-align: center;
  font-family: $general-font;
}

@mixin pageTitleFont {
  z-index: 1;
  color: white;
  box-shadow: none !important;
  @include alignedGeneralFont();
  animation: flicker-cerise 1.5s infinite alternate;
  flex: 0;
}

@mixin brickBackground {
  background: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.8)),
    url("https://i.imgur.com/mnVAkgM.jpg");
  background-repeat: no-repeat;
  background-size: cover;
  height: 80vh;
  flex-direction: column;
  @include centerVerticalHorizontal();
}
