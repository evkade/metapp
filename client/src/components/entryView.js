import React from 'react'; 
import { useState } from 'react';
import './components.css';
import img from './images/brick_wall.jpg';

const EntryView = () => {
    return (
        <div className="entryView">
            <p className="entryView__neonText">
                METAPP
            </p> 
            <div>
                <button>
                    log in
                </button>
                <button>
                    sign up
                </button>
            </div>
        </div>
    );
}

export default EntryView;