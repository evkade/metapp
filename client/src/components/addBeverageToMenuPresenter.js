import React from "react";
import { getCocktailBasedOnName, getCannedBeverageBasedOnName } from "../data/fetchData.ts";
import { AddBeverageToMenu } from "./addBeverageToMenu.js"; // ehm kanske borde passa detta ist fÃ¶r importera ? oklart
import usePromise from '../utils/usePromise.js';

export const AddBeverageToMenuPresenter = () => {
  const [cocktailPromise, setCocktailPromise] = React.useState(undefined);
  const [data, error] = usePromise(cocktailPromise);

  const searchBeverage = (query) => {
    setCocktailPromise(getCannedBeverageBasedOnName(query)); 
  }

  return (
    <AddBeverageToMenu
      searchBeverage={searchBeverage}
      searchResult={data}
    />
  );
};
