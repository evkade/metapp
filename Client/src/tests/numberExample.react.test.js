import React from "react";
import jestRenderer from "react-test-renderer";
import { numberExample } from "../components/numberExample";
import { shallow } from "enzyme";

test("The component should render an div that says 1 as myNumber is 1", () => {
  const component = jestRenderer.create(<numberExample renderNumber={true} myNumber={1} />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test("shallow render works ", () => {
  const component = shallow(<numberExample renderNumber={false} myNumber={1} />);
  expect(component.contains("Random Text"));
});
