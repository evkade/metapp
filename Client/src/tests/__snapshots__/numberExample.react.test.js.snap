// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render an div that says 1 as myNumber is 1 1`] = `
<numberExample
  myNumber={1}
  renderNumber={true}
/>
`;
